File: Exercise_1.java

//Time Complexity - O(n+m)
//Space Complexity - O(1)

class Solution {
    public void merge(int[] nums1, int m, int[] nums2, int n) {

        // using three pointers

        // set 2 pointers to the end of each array

        int p1 = m-1;
        int p2 = n-1;

        //third pointer
        for( int p = m + n - 1; p >= 0; p-- ) {

            //end of array break
            if(p2 < 0) {
                break;
            }

            // moving first pointer case
            if( p1 >= 0 && nums1[p1] > nums2[p2]) {

                nums1[p] = nums1[p1];
                p1--;

            }

            // moving second pointer case
            else  {

                nums1[p] = nums2[p2];
                p2--;

            }


        }


    }
}

File: Exercise_1.cpp
// Time Complexity                              :  O(N)
// Space Complexity                             :  O(1)
// Did this code successfully run on Leetcode   :  Yes
// Any problem you faced while coding this      :  No

#include <bits/stdc++.h> 
using namespace std;  

// https://leetcode.com/problems/remove-duplicates-from-sorted-array-ii/
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        int n = nums.size();
        if(n == 0) return 0;
        int l = 0;
        int count = 1, f = 1, s = 1;
        // using fast and slow pointer technique
        while(f < n) {
            
            if(nums[f] == nums[f-1]) {
                count++;
            } else {
                count = 1;
            }
            
            if(count <= 2) {
                nums[s] = nums[f];
                s++;
            }
            
            f++;
        }
        
        return s;
    }
};